src/main/java/com/twingly/search/Query.java

```diff
@@ -1,94 +1,228 @@
 package com.twingly.search;
 
+import java.text.SimpleDateFormat;
 import java.util.Date;
+import java.util.Objects;
+
+import static com.twingly.search.Constants.SEARCH_QUERY_DATE_FORMAT;
 
 /**
  * The type Query.
  */
 public class Query {
+    private static final SimpleDateFormat QUERY_DATE_FORMAT = new SimpleDateFormat(SEARCH_QUERY_DATE_FORMAT);
     private Date startTime;
     private Date endTime;
-    private String searchPattern;
-    private String documentLanguage;
+    private String searchQuery;
+    private String lang;
+    private String location;
 
     /**
      * Package-level constructor for API usage only. API users should use QueryBuilder instead.
      *
      * @see QueryBuilder
+     * @deprecated since 3.0.0 - use {@link #Query(Date, Date, String, String, String)} instead
      */
+    @Deprecated
     Query() {
         //
     }
 
+    Query(Date startTime, Date endTime, String searchQuery, String lang, String location) {
+        this.startTime = startTime;
+        this.endTime = endTime;
+        this.searchQuery = searchQuery;
+        this.lang = lang;
+        this.location = location;
+    }
+
+    /**
+     * Gets language
+     *
+     * @return language
+     * @since 3.0.0
+     */
+    public String getLang() {
```

Comment:
Ignore my comment :) You can go ahead with `get/setLang`. I got a bit confused as the Java client uses a query builder which none of the other clients does.